rust/kernel/static_key.rs

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: GPL-2.0
+
+// Copyright (C) 2024 Google LLC.
+
+//! Logic for static keys.
+
+use crate::bindings::*;
+
+/// Doc todo
+#[macro_export]
+#[cfg(target_arch = "x86_64")]
+#[cfg(not(CONFIG_HAVE_RUST_ASM_GOTO))]
+macro_rules! static_key_false {
+    ($key:path, $keytyp:ty, $field:ident) => {{
+        let mut output = 1u32;
+
+        core::arch::asm!(
+            r#"
+            1: .byte 0x0f,0x1f,0x44,0x00,0x00
+
+            .pushsection __jump_table,  "aw"
+            .balign 8
+            .long 1b - .
+            .long 3f - .
+            .quad {0} + {1} - .
+            .popsection
+
+            2: mov {2:e}, 0
+            3:
+            "#,
+            sym $key,
+            const ::core::mem::offset_of!($keytyp, $field),
```

Comment:
You probably want to ensure that the type of $key.$field is actually `static_key`.